pkg/component/registrycaches/registry_caches_test.go

```diff
@@ -369,22 +369,23 @@ status: {}
 				Expect(managedResourceSecret.Immutable).To(Equal(ptr.To(true)))
 				Expect(managedResourceSecret.Labels["resources.gardener.cloud/garbage-collectable-reference"]).To(Equal("true"))
 
-				Expect(managedResourceSecret.Data).To(HaveLen(8))
-				dockerConfigSecretName := "registry-docker-io-config-340eea9b"
-				dockerConfigSecret := configSecretYAMLFor(dockerConfigSecretName, "registry-docker-io", "docker.io", configYAMLFor("https://registry-1.docker.io", "336h0m0s", "", ""))
-				Expect(string(managedResourceSecret.Data["secret__kube-system__"+dockerConfigSecretName+".yaml"])).To(Equal(dockerConfigSecret))
-				Expect(string(managedResourceSecret.Data["service__kube-system__registry-docker-io.yaml"])).To(Equal(serviceYAMLFor("registry-docker-io", "docker.io", "https://registry-1.docker.io")))
-				dockerStatefulSet := statefulSetYAMLFor("registry-docker-io", "docker.io", "10Gi", dockerConfigSecretName, nil)
-				Expect(string(managedResourceSecret.Data["statefulset__kube-system__registry-docker-io.yaml"])).To(Equal(dockerStatefulSet))
-				Expect(string(managedResourceSecret.Data["verticalpodautoscaler__kube-system__registry-docker-io.yaml"])).To(Equal(vpaYAMLFor("registry-docker-io")))
+				manifests, err := test.ExtractManifestsFromManagedResourceData(managedResourceSecret.Data)
```

Comment:
I think we can make use of the `NewManagedResourceConsistOfObjectsMatcher`, see example usage in https://github.com/gardener/gardener/blob/1fc2b4953ed6ddd8823a2cb0e14a5014f06b6150/pkg/component/autoscaling/clusterautoscaler/bootstrap_test.go#L49. This matcher already internally handles the brotli extraction. But we can evaluate it separately, I think the current adaptation is good enough.